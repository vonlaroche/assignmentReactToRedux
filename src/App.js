import React, { Component } from 'react';
import Icon from "./Components/Icon/Icon";
import NoteList from "./Components/NoteList/NoteList";
import AddNoteForm from "./Components/AddNoteForm/AddNoteForm";
import './App.css';
import {connect} from "react-redux";


class App extends Component {

  state = {
    // notes: ["Proper Note 1",
    //   "Proper Note 2",
    //   "ðŸ˜€",
    //   "",
    //   "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse rhoncus ante metus, vel consectetur urna mollis ut. Fusce vel massa vestibulum, viverra sem eget, gravida lectus. Aliquam ac dui non."],
    // newNoteValue: "",
    // isHidden: true
  }


  // handleInputChange = event => {
  //   this.setState({ newNoteValue: event.target.value })
  // }

  // handleAddNote = event => {
  //   event.preventDefault();
  //   let noteToAdd = this.state.newNoteValue;
  //   this.setState({ notes: this.state.notes.concat(noteToAdd), newNoteValue: "" });
  // }

  // handleDeleteNote = index => {
  //   let array = [...this.state.notes]; // make a separate copy of the array
  //   array.splice(index, 1);
  //   this.setState({ notes: array });
  // }


  render() {

    return (<div className="App">
      <section className="mainPageSection">

        <div className="sideView">
          <h3>My notes</h3>
          <hr />
          <NoteList notes={this.props.state.notes} wait={2000} onDelete={this.props.handleDeleteNote} />
        </div>

        <div className="dialogWindowView">

          <div className={this.props.state.isHidden ? "quickListView hide" : "quickListView"}>
            <h3 className="listTitle">Append some new notes</h3>
            <div className="list">
              <NoteList notes={this.props.state.notes} wait={1000} onDelete={this.props.handleDeleteNote} />
            </div>

            <AddNoteForm
              onSubmit={event => this.props.handleAddNote(event)}
              value={this.props.newNoteValue}
              onChange={this.props.handleInputChange}
            />
          </div>
          <Icon click={this.props.handleMinMax}></Icon>
        </div>

      </section>
    </div>);
  }
}

const mapStateToProps = state =>{
 return{
   state: state
 }
}

const mapDispatchToProps = dispatch => {
  return {
    handleAddNote: (event) => dispatch({ type:"ADD_NOTE", event:event}),
    handleInputChange: (event) => dispatch({type:"HANDLE_CHANGE", event:event}),
    handleMinMax: () => dispatch({type:"HANDLE_MIN_MAX"}),
    handleDeleteNote: (index) => dispatch({type:"DELETE_NOTE", index:index})
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
