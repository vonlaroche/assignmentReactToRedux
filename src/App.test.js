import React from 'react';
import { create } from "react-test-renderer";
import ReactDOM from 'react-dom';
import App from './App';
import NoteListItem from "./Components/NoteListItem/NoteListItem";
import { tsExternalModuleReference, exportAllDeclaration } from '@babel/types';

const  notesToTest = { 
  validNote: "Proper note", 
  emptyNote: "", 
  tooLongNote: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse rhoncus ante metus, vel consectetur urna mollis ut. Fusce vel massa vestibulum, viverra sem eget, gravida lectus. Aliquam ac dui non.", 
  emojiNote: "ðŸ˜€ðŸ˜€ðŸ˜€"}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("Note List Item component", () => {
  test("it matches the snapshot, proper note, no errors", () => {
    const component = create(<NoteListItem checkNote={notesToTest.validNote}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("Note List Item component", () => {
  test("it matches the snapshot, error, empty note", () => {
    const component = create(<NoteListItem checkNote={notesToTest.emptyNote}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});


describe("Note List Item component", () => {
  test("it matches the snapshot, error, longer than a 100", () => {
    const component = create(<NoteListItem checkNote={notesToTest.tooLongNote}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});


describe("Note List Item component", () => {
  test("it matches the snapshot, error, contains emoji", () => {
    const component = create(<NoteListItem checkNote={notesToTest.emojiNote}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});