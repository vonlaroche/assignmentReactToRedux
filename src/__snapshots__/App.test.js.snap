// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note List Item component it matches the snapshot 1`] = `
Array [
  <li>
    <span />
    <i
      className="fas fa-trash-alt deleteIcon"
    />
  </li>,
  "",
]
`;

exports[`Note List Item component it matches the snapshot, error, contains emoji 1`] = `
Array [
  <li>
    <span />
    <i
      className="fas fa-trash-alt deleteIcon"
    />
  </li>,
  <span
    className="errorSpan"
  >
    Error, contains emojis
  </span>,
]
`;

exports[`Note List Item component it matches the snapshot, error, empty note 1`] = `
Array [
  <li>
    <span />
    <i
      className="fas fa-trash-alt deleteIcon"
    />
  </li>,
  <span
    className="errorSpan"
  >
    Error, note is empty
  </span>,
]
`;

exports[`Note List Item component it matches the snapshot, error, longer than a 100 1`] = `
Array [
  <li>
    <span />
    <i
      className="fas fa-trash-alt deleteIcon"
    />
  </li>,
  <span
    className="errorSpan"
  >
    Error, number of characters is longer than a 100
  </span>,
]
`;

exports[`Note List Item component it matches the snapshot, proper note, no errors 1`] = `
Array [
  <li>
    <span />
    <i
      className="fas fa-trash-alt deleteIcon"
    />
  </li>,
  "",
]
`;
